Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -58,34 +58,24 @@ protected function __construct(string $identifier)
     /**
      * Apply an event to the current aggregate root
      *
-     * If the event aggregate identifier and name is not set the event
-     * is automatically set with the current aggregate identifier
-     * and name.
+     * If the event is an instance of AggregateEventInterface, the aggregate identifier will be set
      *
      * @param EventInterface $event
-     * @param array $metadata
-     * @return void
      * @api
      */
-    final public function recordThat(EventInterface $event, array $metadata = [])
+    final public function recordThat(EventInterface $event)
```

Comment:
> The logic for versioning (the events) belongs to the aggregate

I don't think that's related. The version is still controlled from the repository

> occuredAt timestamp gets shifted to the time the event is started processing for persisting

That's true, but it usually happens *right* afterwards. If you need higher precision the timestamp most probably is part of your domain and thus of the event payload

> other aggregate relevant metadata can no longer be specified by an aggregate implementation itself

Like? Again, if it's aggregate-relevant it should not be metadata IMO